src/System.Data.SqlClient/tests/ManualTests/README.md

```diff
@@ -1,6 +1,32 @@
 # SqlClient Manual Tests
 
-These tests require dedicated test servers, so they're designed to be run manually using a custom set of connection strings. These connection strings should be added as environment variables: TEST\_NP\_CONN\_STR & TEST\_TCP\_CONN\_STR. TEST\_NP\_CONN\_STR is a named pipes connection string to the test server, and TEST\_TCP\_CONN\_STR is a TCP connection string. Each protocol can be specified in the Server name parameter. These tests also assume the sample database "Northwind" exists in the target server. This sample database can be found [here](https://msdn.microsoft.com/en-us/library/mt710790.aspx).
+These tests require dedicated test servers, so they're designed to be run manually using a custom set of connection strings. 
 
-Instructions for running tests: [Unix](https://github.com/dotnet/corefx/blob/master/Documentation/building/cross-platform-testing.md) and [Windows](https://github.com/dotnet/corefx/blob/master/Documentation/building/windows-instructions.md).
-Documentation for connection string parameters: [SqlConnection.ConnectionString](https://msdn.microsoft.com/en-us/library/system.data.sqlclient.sqlconnection.connectionstring.aspx).
\ No newline at end of file
+## Prerequisites
+
+- CoreFX building. you need to be able to do a successful `build -allconfigurations` and run the standard tests, [Unix](https://github.com/dotnet/corefx/blob/master/Documentation/building/cross-platform-testing.md) or [Windows](https://github.com/dotnet/corefx/blob/master/Documentation/building/windows-instructions.md)
+
+- an [MS SQL Server](https://www.microsoft.com/en-us/sql-server/sql-server-editions-express) (any edition) 2012 or later that you can connect to with tcp and named pipes, 
+
+  **N.B**. if you want to run the EFCore tests it should be a dedicated instance because they create a lot of databases.
+
+- The  [Northwind Sample Database](https://msdn.microsoft.com/en-us/library/mt710790.aspx)
+
+- The [UDT Test Database](https://github.com/dotnet/corefx/tree/master/src/System.Data.SqlClient/tests/ManualTests/createUdtTestDb_corefx.sql) 
+
+- TCP and Named Pipe [connection strings](https://msdn.microsoft.com/en-us/library/system.data.sqlclient.sqlconnection.connectionstring.aspx) to your instance with Northwind set as the initial catalog
+
+
+
+## Running All Tests
+
+1. run `build src\System.Data.SqlClient -allconfigurations` and make sure the builds all work
```

Comment:
1 & 4 can be combined into a single statement
`dotnet msbuild .\src\System.Data.SqlClient\tests\ManualTests\System.Data.SqlClient.ManualTesting.Tests.csproj /t:RebuildAndTest`
This would build and run only the `Manual Tests` as compared to running all tests with command `build src\System.Data.SqlClient -debug /p:ForceRunTests=true`

Reply:
If you combine then you get test infrastructure failures:
```
  Executing in E:\Programming\csharp7\corefx\artifacts\bin\tests\System.Data.SqlClient.Tests\netcoreapp-Windows_NT-Debug-x64\
  ----- start 22:39:13.28 ===============  To repro directly: =====================================================
  pushd E:\Programming\csharp7\corefx\artifacts\bin\tests\System.Data.SqlClient.Tests\netcoreapp-Windows_NT-Debug-x64\
  popd
  ===========================================================================================================
  A JSON parsing exception occurred in [E:\Programming\csharp7\corefx\artifacts\bin\tests\System.Data.SqlClient.Tests\netcoreapp-Windows_NT-Debug-x64\xunit.console.runtimeconfig.json]: * Line 1, Column 2 Syntax error: Malformed token
  Invalid runtimeconfig.json [E:\Programming\csharp7\corefx\artifacts\bin\tests\System.Data.SqlClient.Tests\netcoreapp-Windows_NT-Debug-x64\xunit.console.runtimeconfig.json] [E:\Programming\csharp7\corefx\artifacts\bin\tests\System.Data.SqlClient.Tests\netcoreapp-Windows_NT-Debug-x64\xunit.console.runtimeconfig.dev.json]
  ----- end 22:39:13.31 ----- exit code -2147450733 ----------------------------------------------------------
```
Running the specific build you want to test one at a time works.

Reply:
Here is my console output for the command `dotnet msbuild .\src\System.Data.SqlClient\tests\ManualTests\System.Data.SqlClient.ManualTesting.Tests.csproj /t:RebuildAndTest`
```
PS D:\dotNetCore\corefx> dotnet msbuild .\src\System.Data.SqlClient\tests\ManualTests\System.Data.SqlClient.ManualTesting.Tests.csproj /t:RebuildAndTest
Microsoft (R) Build Engine version 15.9.20+g88f5fadfbe for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Address -> D:\dotNetCore\corefx\artifacts\bin\Address\netstandard-AnyOS-Debug\Address.dll
  Utf8String -> D:\dotNetCore\corefx\artifacts\bin\Utf8String\netstandard-AnyOS-Debug\Utf8String.dll
  Shapes -> D:\dotNetCore\corefx\artifacts\bin\Shapes\netstandard-AnyOS-Debug\Shapes.dll
  Circle -> D:\dotNetCore\corefx\artifacts\bin\Circle\netstandard-AnyOS-Debug\Circle.dll
  System.Data.SqlClient.ManualTesting.Tests -> D:\dotNetCore\corefx\artifacts\bin\System.Data.SqlClient.ManualTesting.Tests\netcoreapp-AnyOS-Debug\System.Data.SqlClient.ManualTesting.Tests.dll
  Using D:\dotNetCore\corefx\artifacts\bin\testhost\netcoreapp-Windows_NT-Debug-x64 as the test runtime folder.
  Executing in D:\dotNetCore\corefx\artifacts\bin\tests\System.Data.SqlClient.ManualTesting.Tests\netcoreapp-Windows_NT-Debug-x64\
  ----- start 15:13:58.80 ===============  To repro directly: =====================================================
  pushd D:\dotNetCore\corefx\artifacts\bin\tests\System.Data.SqlClient.ManualTesting.Tests\netcoreapp-Windows_NT-Debug-x64\
  D:\dotNetCore\corefx\artifacts\bin\testhost\netcoreapp-Windows_NT-Debug-x64\dotnet.exe xunit.console.dll System.Data.SqlClient.ManualTesting.Tests.dll -xml testResults.xml -nologo -notrait category=nonnetcoreapptests -notrait category=nonwindowstests -notrait category=failing -notrait category=Outerloop
  popd
  ===========================================================================================================
    Discovering: System.Data.SqlClient.ManualTesting.Tests (method display = ClassAndMethod, method display options = None)
    Discovered:  System.Data.SqlClient.ManualTesting.Tests (found 206 test cases)
    Starting:    System.Data.SqlClient.ManualTesting.Tests (parallel test collections = on, max threads = 8)
```

Reply:
So running the above command will only build and test on platform based on the RID picked up automatically, which is `netcoreapp-Windows_NT-Debug-x64` in the case above. `allconfigurations` will not work, since this is not building all configurations. The thing is that `build src\System.Data.SqlClient` chooses the `csproj` in the folder `src\System.Data.SqlClient` which means it is running all the other tests in `SqlClient` along with the manual tests.

Reply:
> Since it's just supposed to get people started I wanted to break the build and test steps out individually.

Fair enough, then the command above can be broken into the following and it only builds and tests `System.Data.SqlClient.ManualTesting.Tests.csproj`
-  `dotnet msbuild .\src\System.Data.SqlClient\tests\ManualTests\System.Data.SqlClient.ManualTesting.Tests.csproj /t:Rebuild`
- `dotnet msbuild .\src\System.Data.SqlClient\tests\ManualTests\System.Data.SqlClient.ManualTesting.Tests.csproj /t:Test`

Reply:
I've tweaked it a bit. See if that's better. I also recently found you don't need to use msbuild to run a specific test, you can do it straight from build.

Reply:
You're right, it does. I was also crediting linux users with enough intelligence to realise that the shell script needed to be executable if they're trying to do development work. I've changed it anyway.